### ISSUE TITLE
Buggy mode: cursor movement becomes slow

### ISSUE THREAD

**Issue by olcc** [geany/geany#1940]
Hello,

When using Geany, after some time or some action that I could not identify, Geany goes into "buggy mode":
- moving the cursor in the text becomes significantly slower,
- the vertical scroll bar repaints about every second even when I do not move or do anything (I can see a slight blink of the scrollbar when it repaints) (this happens when Geany window has the focus).

Details below.

Olivier

---
Debian: testing
Window manager: Xfce

$ geany -V
geany 1.33 (construit le Apr 16 2018 avec GTK 3.22.29, GLib 2.56.1)

Disable plugins? -> problem still occurs
Window full screen? -> problem still occurs
Output to consol? -> no text is displayed on the console when the problem occurs

CPU usage in buggy mode? When moving the cursor with the Up/Down keys inside the text of the Geany window, without causing lines to scroll...

|             |  No movement   |  Up/Down keys |
| --------- | :---------------------: | :--------------------: |
| geany  |         ∼ 0 %        |        ∼ 14 %      |
| Xorg    |         ∼ 1 %        |         ∼ 70 %     |

Hello Geany, do you really need all that CPU for moving a cursor in the text???

**Comment by elextr** [comment]
> Hello Geany, do you really need all that CPU for moving a cursor in the text???

No, it probably needs it to respond to all the messages X at 70% is sending.

**Comment by elextr** [comment]
Without identifying the trigger its unlikely that devs will be able to reproduce this, so the best you can do is keep trying to find the trigger.

BTW when you say "disable plugins" does that mean restarting Geany with ALL plugins disabled and leaving them disabled?  And the problem still occurs?

There is a known issue with the scope plugin that triggers a GTK bug, but that usually causes 100% Geany CPU IIUC, so its probably not that.

**Comment by olcc** [comment]
I tested with all plugins disabled and restarting Geany in this situation -> problem still occurs
I do not have a "scope plugin".
